Doc: Changelog for 0.5.0 release (#104)
OpenTTD/nml: docs/changelog.txt

0.5.0 (2020-04-26)
------------------------------------------------------------------------
- Feature: Support for RoadTypes (a.k.a NRT)

> - Feature: Industries now support up to 16 accepted cargos and 16 produced cargos

Reviewer: Looking at the commit, this isn't nearly enough information or even an accurate summary.

It doesn't only *add* support for more cargos, it removes the previous syntax and properties in a way that will break all industry-related grfs.

The changelog needs to say that, and either list the required changes or link to somewhere that does.

(IMO such a big change should really have had a deprecation period, but too late now),

Reviewer 2: > (IMO such a big change should really have had a deprecation period, but too late now),

I initially left support for the old syntax in, but then was told that the NML way is to remove all traces of the old way of doing things when a new and better way is introduced.

Agree that the changelog should at least warn that the syntax has changed and all projects that define industries need to be updated, or stay on the old version.

0.5.0 (2020-04-26)
------------------------------------------------------------------------

> 

Reviewer 3: extra newline

- Feature: Add ANIM_TRIGGER_APT_AIRPLANE_LANDS (follows OpenTTD #7182)
- Feature: Add station var nearby_tile_grfid (0x6A)
- Feature: PUSH/POP_COLOUR codes for strings
- Feature: Debug_print for GRF parameters
- Feature: Allow PLY to generate parsing/lexing tables (improves performance); regeneration can be forced with nmlc -R
- Feature: Visual Studio Editor Script
- Change: improved colouring of nmlc errors and warnings (where shell/terminal supports this)
- Change: Make sprite encoder warnings more informative (#89)
- Add: nmlc -D option to enable debug mode for parser
- Fix #36: Global variable traffic_side was inconsistent with global parameter traffic_side
- Fix #39: Add compatibility with >=pillow-7.0.0
- Fix #52: time.clock() has been removed in python 3.8
- Fix #57: Check coherency of GRF parameters limits
- Fix: Add missing parameter in industry and airport tiles in RandomAction2
- Fix: Convert some code flows to context managers (eliminates 'too many open files' error on macOS with larger grfs)
- Fix: LOAD_TEMP() 'value' attribute is empty
- Codechange: Rewrite setup.py
- Codechange: Install_requires in setup.py
- Codechange: pyinstaller spec file

> - Codechange: Update nml version detection to support Git using similar approach to OpenTTD

Reviewer 3: keep this codechange, i don't see the need in keeping any others

0.5.0 (2020-04-26)
------------------------------------------------------------------------

- Feature: Industries now support up to 16 accepted cargos and 16 produced cargos
- Feature: Houses now support up to 16 accepted cargos
- Feature: Support for RoadTypes (a.k.a NRT)
- Feature: Allow use of switches and random switches as procedures, reducing repetition of switches
- Feature: Increase railtype ID range from 16 to 64, to support 64 railtypes
- Feature: Railtype flags RAILTYPE_FLAG_ALLOW_90DEG and RAILTYPE_FLAG_DISALLOW_90DEG for 90 degree curves
- Feature: Railtype flags RAILTYPE_FLAG_HIDDEN and RAILTYPE_FLAG_PRECOMBINED for precombined spriteset type
- Feature: Add ANIM_TRIGGER_APT_AIRPLANE_LANDS (follows OpenTTD #7182)
- Feature: Add station var nearby_tile_grfid (0x6A)
- Feature: PUSH/POP_COLOUR codes for strings

> - Feature: Support industry vars production_rate (0x70) and transported_last_month_pct (0x71)

Reviewer: Reorder so this is with the 16-cargo industry change above? IIRC this was needed to support production rates for all those cargoes.

- Feature: Debug_print for GRF parameters

> - Feature: nmlc option to enable debug mode for parser

Reviewer: Add (`-D`) to this line?

Reviewer 3: I'd probably say it should be an "Add: " as well, it's not exactly a "feature"

Reviewer: Yeah, end-users don't care, this only makes nml developers' life easier.

> - Feature: nmlc option to force regeneration of parser tables

Reviewer: This and "Codechange: Allow PLY to generate parsing/lexing tables (performance improvement)" should probably be rolled together. Something like:

> - Feature: Allow PLY to generate parsing/lexing tables (improves performance); regeneration can be forced with `nmlc -R`.

- Feature: Add TTDPatchFlag 256_persistent_registers (0x80)
- Change: improved colouring of nmlc errors and warnings (where shell/terminal supports this)
- Change: Make sprite encoder warnings more informative (#89)
- Fix #36: Global variable traffic_side was inconsistent with global parameter traffic_side
- Fix #39: Add compatibility with >=pillow-7.0.0

> - Fix #58: Loosen Pillow version constraint

Reviewer 3: remove

> - Fix #52: time.clock() has been removed in python 3.8

Reviewer 3: remove

Reviewer: Why? This incompatibility was a specific reason the Debian packager was asking for a prompt release of 0.5.0.

Reviewer 3: That's true. Perhaps
```suggestion
- Fix #52: Compatibility with python 3.8
```

> - Fix #57: Check coherency of GRF parameters limits

Reviewer 3: fixes should describe what has been fixed, not what the fix was
```suggestion
- Fix #57: GRF parameters could be given negative values
```

> - Fix 014febc: Wrong link to LICENSE file

Reviewer 3: remove

> - Fix 78cd84c: Restore 'setup.py install' usability on windows

Reviewer 3: should not refer to commit hashes

Reviewer: Both of the fixes above are for problems introduced since 0.4.5, so they should be removed from the changelog.

- Fix: Add missing parameter in industry and airport tiles in RandomAction2

> - Fix: Convert some code flows to context managers (eliminates 'too many open files' error on macOS with larger grfs)

Reviewer 3: ```suggestion
- Fix: Properly clean up resource handles (eliminates 'too many open files' error on macOS with larger GRFs)
```

- Fix: LOAD_TEMP() 'value' attribute is empty

> - Fix: Update various sprite counts

Reviewer 3: again, find why this was done, not what

Author: It's to support newly added UI + tram sprites for NRT, it's not really relevant to end user

> - Codechange: Visual Studio Editor Script

Reviewer 3: in general, all codechanges should be removed

> - Codechange: Visual Studio Editor Script

Reviewer 3: this should be a feature though

- Codechange: Install_requires in setup.py
- Codechange: pyinstaller spec file
- Codechange: Allow PLY to generate parsing/lexing tables (performance improvement)

> - Codechange: Avoid duplicating rail/road/tramtype table code

Reviewer: That code didn't exist in the last release, so not really a change for these purposes.

- Codechange: Update nml version detection to support Git using similar approach to OpenTTD
- Codechange: Mechanical cleanups of Python code
- Codechange: Remove cx_Freeze
- Codechange: Rewrite setup.py
- Codechange: Simplify Pillow imports
- Codechange: Simplify Pillow version detection
- Codechange: Move license file to root folder

> - Doc: Update README

Reviewer 3: remove

- Doc: 0.3 changed 'shorten_vehicle' to 'length' - this wasn't updated in one place in train example nml
- Doc: Add example train wagon showing sprite stack use

> - Doc: Fix 404 on pillow install instructions link

Reviewer 3: remove

- Doc/Fix: Update the stated version requirements

> - Cleanup: Industry vars production_rate_1 (0x8E) and production_rate_2 (0x8F) are deprecated and replaced by new parametric var production_rate (0x70)

Reviewer: Do the old ones still work? If not this should be much higher.

Reviewer 3: Indeed, if it's not an actual removal, it's not a "cleanup", rather a "change"

Author: +1 this will cause some people's compiles to fail;  I really think this has to be out-of-order and follow directly from the 'Feature: production_rate 0x70 blah blah' line.
